src/stream.c

```diff
@@ -741,7 +744,14 @@ void flush_streamb(streams *sid, unsigned char *buf, int rlen, int64_t ctime)
 	int i, rv = 0, blen, len;
 
 	if (sid->type == STREAM_HTTP)
+	{
 		rv = sockets_write(sid->rsock_id, buf, rlen);
+		if (sid->start_streaming == 0)
```

Comment:
You should find a new code place where you can set sid->start_streaming to 1 that reaches all code paths and you do not repeat the same code block. (basically set start_streaming to 1 in 1 location instead of 3).

Reply:
Other than those comments everything looks good.

Reply:
> You should find a new code place where you can set sid->start_streaming to 1 that reaches all code paths and you do not repeat the same code block. (basically set start_streaming to 1 in 1 location instead of 3).

No. As I commented, the `start_streaming`to 1 in 1 is a simple error (copy&paste from my other PR). The correct place is here. The start streaming is detected for HTTP here, as it's when the data (of the streaming) is started to be writed in the socket.

Reply:
> Other than those comments everything looks good.

Please, compile with this patch and check the logs with HTTP streaming.
I'm sure you see it good. So, please commit it! :wink: